### PR TITLE
(fix) O3-3614: Fix handling of the priority field in the lab results form

### PR DESCRIPTION
**Pull request by jwnasambu**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

Presently, the value of the priority field in the Lab results form does not getting added included in the payload when creating, modifying or discontinuing a test order. This PR fixes that by adding the urgency field to the payload.

## Screenshots

https://github.com/user-attachments/assets/e26537be-7dd5-4d93-8310-684ffc15d5f0

## Related Issue
https://openmrs.atlassian.net/browse/O3-3614

### CODE REVIEW

File: packages/esm-patient-medications-app/src/add-drug-order/drug-search/drug-search.resource.tsx
[the remark this answers is not in the document]
Comment (jwnasambu) [reply packages/esm-patient-medications-app/src/add-drug-order/drug-search/drug-search.resource.tsx:1]: Kindly is it on test app or orders app?

[the remark this answers is not in the document]
Comment (denniskigen) [reply packages/esm-patient-medications-app/src/add-drug-order/drug-search/drug-search.resource.tsx:1]: Orders, I mean

**Comment by denniskigen** [comment]
From taking a cursory look at this, it looks like we need to do is add an `urgency` property to the [prepLabOrderPostData](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-tests-app/src/lab-orders/api.ts#L79) function in `api.ts` that maps to `order.urgency`.

File: packages/esm-patient-tests-app/src/lab-orders/api.ts [hunk 1]
```diff
@@ -93,6 +95,7 @@ export function prepLabOrderPostData(
       instructions: order.instructions,
       orderReason: order.orderReason,
       accessionNumber: order.accessionNumber,
+      urgency: urgency,
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/lab-orders/api.ts:98]:
```suggestion
      urgency: order.urgency,
```

File: packages/esm-patient-tests-app/src/lab-orders/api.ts [hunk 2]
```diff
@@ -81,6 +81,8 @@ export function prepLabOrderPostData(
   patientUuid: string,
   encounterUuid: string | null,
 ): OrderPost {
+  const urgency = order.urgency ?? 'ROUTINE';
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/lab-orders/api.ts:84]:
```suggestion
```

`urgency` is a required field in the form, so we shouldn't need to assign a default value.

File: packages/esm-patient-tests-app/src/lab-orders/api.ts [hunk 3]
```diff
@@ -107,6 +110,7 @@ export function prepLabOrderPostData(
       orderReason: order.orderReason,
       previousOrder: order.previousOrder,
       accessionNumber: order.accessionNumber,
+      urgency: urgency,
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/lab-orders/api.ts:113]:
```suggestion
      urgency: order.urgency,
```

File: packages/esm-patient-tests-app/src/lab-orders/api.ts [hunk 4]
```diff
@@ -120,6 +124,7 @@ export function prepLabOrderPostData(
...
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/lab-orders/api.ts:127]:
```suggestion
      urgency: order.urgency,
```

File: packages/esm-patient-common-lib/src/orders/types.ts [hunk 5]
```diff
@@ -46,6 +47,7 @@ export interface OrderBasketItem {
 }
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/orders/types.ts:48]: export type OrderUrgency = 'ROUTINE' | 'STAT' | 'ON_SCHEDULED_DATE';

File: packages/esm-patient-common-lib/src/orders/types.ts [hunk 6]
```diff
@@ -207,7 +209,7 @@ export interface LabOrderBasketItem extends OrderBasketItem {
     label: string;
     conceptUuid: string;
   };
-  urgency?: string;
+  urgency: string;
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/orders/types.ts:212]:
```suggestion
  urgency: OrderUrgency;
```

File: packages/esm-patient-common-lib/src/orders/types.ts [hunk 7]
```diff
@@ -46,6 +47,7 @@ export interface OrderBasketItem {
 }
 
 export interface OrderPost {
+  urgency: string;
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/orders/types.ts:50]:
```suggestion
  urgency: OrderUrgency;
```

File: packages/esm-patient-common-lib/src/orders/postOrders.ts
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/orders/postOrders.ts:1]: We shouldn't need any of the changes in this file.

Comment (jwnasambu) [reply packages/esm-patient-common-lib/src/orders/postOrders.ts:1]: @denniskigen Thanks for the review.

File: packages/esm-patient-orders-app/src/utils/index.ts [hunk 8]
```diff
@@ -63,7 +63,7 @@ export function buildMedicationOrder(order: Order, action?: OrderAction): DrugOr
       value: order.quantityUnits?.display,
       valueCoded: order.quantityUnits?.uuid,
     },
-  };
+  } as DrugOrderBasketItem;
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/utils/index.ts:66]:
```suggestion
  };
```

**Review by denniskigen — COMMENTED** [review]
Couple more things, but this is looking good.

**Review by denniskigen — APPROVED** [review]
Logic LGTM. Have you retested locally and found that it works?

**Comment by jwnasambu** [comment]
> Logic LGTM. Have you retested locally and found that it works?

@denniskigen Yes it's working as expected and I am working on test failures.

File: packages/esm-patient-common-lib/src/orders/types.ts [hunk 9]
```diff
@@ -207,7 +210,7 @@ export interface LabOrderBasketItem extends OrderBasketItem {
     label: string;
     conceptUuid: string;
   };
-  urgency?: string;
+  urgency: OrderUrgency;
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/orders/types.ts:213]:
```suggestion
  urgency?: OrderUrgency;
```

File: packages/esm-patient-common-lib/src/orders/types.ts [hunk 10]
```diff
@@ -45,7 +45,10 @@ export interface OrderBasketItem {
   accessionNumber?: string;
 }
 
+export type OrderUrgency = 'ROUTINE' | 'STAT' | 'ON_SCHEDULED_DATE';
+
 export interface OrderPost {
+  urgency: OrderUrgency;
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/orders/types.ts:51]:
```suggestion
  urgency?: OrderUrgency;
```

**Review by denniskigen — COMMENTED** [review]
Let's try making `urgency` an optional prop instead.

**Comment by denniskigen** [comment]
To fix the type issue conclusively, you'll need to modify [this file](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-tests-app/src/lab-orders/add-lab-order/lab-order.ts):

Add a type annotation for `PriorityOption`:

```tsx
type PriorityOption = {
  value: OrderUrgency;
  label: string;
};
```

And then annotate the [priorityOptions](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-tests-app/src/lab-orders/add-lab-order/lab-order.ts#L12) object with it.

File: packages/esm-patient-tests-app/src/lab-orders/add-lab-order/lab-order.ts [hunk 11]
```diff
@@ -3,8 +3,15 @@ import { type TestType } from './useTestTypes';
 
 type LabOrderRequest = Pick<LabOrderBasketItem, 'action' | 'testType'>;
 
+type OrderUrgency = 'ROUTINE' | 'STAT' | 'ON_SCHEDULED_DATE';
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/lab-orders/add-lab-order/lab-order.ts:6]: Can we import it from `@openmrs/esm-patient-common-lib` instead?

```suggestion
```

**Comment by denniskigen** [comment]
LGTM. Thanks, @jwnasambu!

**Pull request merged**